openmetadata-ui/src/main/resources/ui/src/pages/signup/index.test.tsx

```diff
@@ -0,0 +1,331 @@
+    const emailLabel = getByTestId('email-label');
+    const emailInput = getByTestId('email-input');
+    const selectTeamLabel = getByTestId('select-team-label');
+    const createButton = getByTestId('create-button');
+    const loadingContent = await queryByTestId('loading-content');
+
+    expect(logo).toBeInTheDocument();
+    expect(heading).toBeInTheDocument();
+    expect(form).toBeInTheDocument();
+    expect(fullNameLabel).toBeInTheDocument();
+    expect(fullNameInput).toBeInTheDocument();
+    expect(usernameLabel).toBeInTheDocument();
+    expect(usernameInput).toBeInTheDocument();
+    expect(emailLabel).toBeInTheDocument();
+    expect(emailInput).toBeInTheDocument();
+    expect(selectTeamLabel).toBeInTheDocument();
+    expect(createButton).toBeInTheDocument();
+    expect(loadingContent).toBeNull();
+
+    await act(async () => {
+      form.onsubmit = mockSubmitHandler;
+
+      fireEvent.submit(form);
+
+      expect(mockSubmitHandler).toBeCalledTimes(1);
+    });
+  });
+
+  it('Handlers in forms for change and submit should work properly', async () => {
+    AppState.default.newUser = {
+      name: 'Sample Name',
+      email: 'sample123@sample.com',
+      picture: 'Profile Picture',
+    };
+    AppState.default.updateUserPermissions = jest.fn();
+    AppState.default.updateUserDetails = jest.fn();
+
+    (createUser as jest.Mock).mockImplementationOnce(() =>
+      Promise.resolve(undefined)
+    );
+
+    const { getByTestId } = render(<Signup />);
+
+    const form = getByTestId('create-user-form');
+    const fullNameInput = getByTestId('full-name-input');
+    const usernameInput = getByTestId('username-input');
+    const emailInput = getByTestId('email-input');
+
+    expect(form).toBeInTheDocument();
+    expect(fullNameInput).toHaveValue('Sample Name');
+    expect(usernameInput).toHaveValue('sample123');
+    expect(emailInput).toHaveValue('sample123@sample.com');
+
+    fullNameInput.onchange = mockChangeHandler;
+    usernameInput.onchange = mockChangeHandler;
+    emailInput.onchange = mockChangeHandler;
+
+    await act(async () => {
+      fireEvent.change(fullNameInput, {
+        target: { name: 'displayName', value: 'Fname Mname Lname' },
+      });
+
+      fireEvent.change(usernameInput, {
+        target: { name: 'displayName', value: 'mockUserName' },
+      });
+      fireEvent.change(emailInput, {
+        target: { name: 'displayName', value: 'sample@sample.com' },
+      });
+
+      expect(mockChangeHandler).toBeCalledTimes(3);
+
+      form.onsubmit = mockSubmitHandler;
+
+      fireEvent.submit(form);
+
+      expect(mockSubmitHandler).toBeCalledTimes(1);
+    });
+  });
+
+  it('Error should be thrown if createUser API fails', async () => {
+    AppState.default.newUser = {
+      name: 'Sample Name',
+      email: 'sample123@sample.com',
+      picture: 'Profile Picture',
+    };
+    AppState.default.updateUserPermissions = jest.fn();
+    AppState.default.updateUserDetails = jest.fn();
```

Comment:
it is very repetitive, can we handle it in a better way?

Reply:
Done ✅